napari/layers/points/points.py

```diff
@@ -380,6 +588,126 @@ def edge_width(self, edge_width: Union[None, float]) -> None:
+
+        else:
+            transformed_color = transform_color_with_defaults(
+                num_entries=len(self.data),
+                colors=edge_color,
+                elem_name="edge_color",
+                default="white",
+            )
+            self._edge_color = normalize_and_broadcast_colors(
+                len(self.data), transformed_color
+            )
+            self.edge_color_mode = ColorMode.DIRECT
+            self._edge_color_property = ''
+
+            self.events.edge_color()
+            self.events.highlight()
+
+    @property
+    def edge_color_cycle(self):
+        """Union[list, np.ndarray, cycle]:  Color cycle for edge_color. Can be a list of colors or a cycle of colors"""
+        return self._edge_color_cycle
+
+    @edge_color_cycle.setter
+    def edge_color_cycle(
+        self, edge_color_cycle: Union[list, np.ndarray, cycle]
+    ):
+        if isinstance(edge_color_cycle, cycle):
+            self._edge_color_cycle = edge_color_cycle
+        else:
+            self._edge_color_cycle = transform_color_cycle(
+                color_cycle=edge_color_cycle,
+                elem_name="edge_color_cycle",
+                default="white",
+            )
+        if self._edge_color_mode == ColorMode.CYCLE:
+            self.refresh_edge_color()
+
+    @property
+    def edge_colormap(self):
+        """(str, vispy.color.Colormap) colormap to be applied to a property to set edge_color"""
+        return self._edge_colormap_name, self._edge_colormap
+
+    @edge_colormap.setter
+    def edge_colormap(self, colormap: Union[str, Colormap]):
+        self._edge_colormap = get_colormap(colormap)
+        if isinstance(colormap, str):
+            self._edge_colormap_name = colormap
+        else:
+            self._edge_colormap_name = 'unknown_colormap'
+
+    @property
+    def edge_contrast_limits(self):
+        """None, (float, float) : clims for mapping the edge_color colormap property to 0 and 1"""
+        return self._edge_contrast_limits
+
+    @edge_contrast_limits.setter
+    def edge_contrast_limits(
+        self, contrast_limits: Union[None, Tuple[float, float]]
+    ):
+        self._edge_contrast_limits = contrast_limits
+
+    def refresh_edge_color(self, update_color_mapping: bool = True):
+        """ Calculate and update edge color if using a cycle or color map
+
+            Parameters
+            ----------
+            update_color_mapping : bool
+                If set to True, the function will recalculate the color cycle map or colormap
+                (whichever is being used). If set to False, the function will use the current color cycle map or
+                color map. For example, if you are adding/modifying points and want them to be colored with the same
+                mapping as the other points (i.e., the new points shouldn't affect the color cycle map or colormap),
+                set update_color_mapping=False. Default value is True.
+        """
```

Comment:
This docstring isn't properly formatted....

Reply:
Perhaps `refresh_edge_color` and `refresh_face_color` could be combined into a single function? At a minimum... It would be great if we could somehow avoid these functions, maybe using the `attrs` library or similar? I have no experience here, I just think it's not a great user experience.

Happy to punt on this, btw, as this PR is already great and insanely useful, but I just wanted to flag this as an issue. And I do think even in keeping the function(s) they can be simplified.

Reply:
> This docstring isn't properly formatted....

Fixed!

Reply:
> Perhaps refresh_edge_color and refresh_face_color could be combined into a single function? At a minimum... It would be great if we could somehow avoid these functions, maybe using the attrs library or similar? I have no experience here, I just think it's not a great user experience.

I think this is a good point. I will respond to code duplication and attr type comments together at the bottom.